components/admin/reports/GenerateReportModal.tsx

```diff
@@ -47,26 +47,51 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
   const [endDate, setEndDate] = useState('');
   const [step, setStep] = useState(GenerateReportStep.SelectColumns);
 
-  // Checkbox states
-  const [selectAll, setSelectAll] = useState(false);
-  const [userID, setUserID] = useState(false);
-  const [applicantName, setApplicantName] = useState(false);
-  const [applicantDoB, setApplicantDoB] = useState(false);
-  const [APPNumber, setAPPNumber] = useState(false);
-  const [applicationDate, setApplicationDate] = useState(false);
-  const [paymentMethod, setPaymentMethod] = useState(false);
-  const [feeAmount, setFeeAmount] = useState(false);
-  const [donationAmount, setDonationAmount] = useState(false);
-  const [totalAmount, setTotalAmount] = useState(false);
+  const [selectedColumns, setSelectedColumns] = useState<Set<ApplicationsReportColumn>>(new Set());
+
+  // A function that returns an event-handler function that adds/removes the column from selectedColumns
```

Comment:
I included this comment as a clarification lol so we can remove this, it's pretty clear what the below function does